### PR TITLE
introduce kubectl plugin

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

**Review by CharlyF — CHANGES_REQUESTED** [review]
From a usage standpoint there are a few things that I think we need to address before merging. 
For instance, there is no prompt if you don't have a subcommand in the dryrun command or if there are no WPAs matching the labels:

```
➜  watermarkpodautoscaler git:(feat/wpa-cli) ./bin/kubectl-wpa dry-run
➜  watermarkpodautoscaler git:(feat/wpa-cli)
[...]
➜  watermarkpodautoscaler git:(feat/wpa-cli) ✗ ./bin/kubectl-wpa dry-run list -l test=a
➜  watermarkpodautoscaler git:(feat/wpa-cli) ✗
```
We should output something with cobra for the incomplete command and a warning "no matching WPAs".

Then, I think the enable and disable commands need an extra step, or a warning. Otherwise it can have too broad of an impact.
```
➜  watermarkpodautoscaler git:(feat/wpa-cli) ./bin/kubectl-wpa dry-run disable
WatermarkPodAutoscaler 'default/example1-watermarkpodautoscaler' dry-run option is now false
WatermarkPodAutoscaler 'default/example2-watermarkpodautoscaler' dry-run option is now false
WatermarkPodAutoscaler 'default/example3-watermarkpodautoscaler' dry-run option is now false
```
and as I did not know the previous state this could be an issue to rollback. 
I suggest using `--all` or "you are about to diable/enable X WPAs, you sure?" to make the command very explicit.

I also noticed that the label selector is not working for the enable/disable subcommand, only for the list:
```
➜  watermarkpodautoscaler git:(feat/wpa-cli) ./bin/kubectl-wpa dry-run enable -l team=aaa
Error: the watermarkpodautoscaler name or label-selector is required
➜  watermarkpodautoscaler git:(feat/wpa-cli) ./bin/kubectl-wpa dry-run list -l team=aaa
WatermarkPodAutoscaler 'default/example1-watermarkpodautoscaler' dry-run option is: false
WatermarkPodAutoscaler 'default/example3-watermarkpodautoscaler' dry-run option is: false
```
This is because of an incorrect if condition in validate, as per my suggestion.

File: cmd/kubectl-wpa/app/dryrun/dryrun.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+	if len(args) > 0 {
+		o.userWPAName = args[0]
+	}
+
+	return nil
+}
+
+// validate ensures that all required arguments and flag values are provided.
+func (o *dryrunOptions) validate() error {
+	if len(o.args) < 1 && o.labelSelector != "" {
```
[the remark this answers is not in the document]
Comment (clamoriniere) [reply cmd/kubectl-wpa/app/dryrun/dryrun.go:194]: good catch

[the remark this answers is not in the document]
Comment (clamoriniere) [reply cmd/kubectl-wpa/app/dryrun/dryrun.go]: hi @CharlyF 
I have addressed all your comments, and have implemented some new functionality to ease "revert" changes.

you can find a new doc page dedicated to the plugin: https://github.com/DataDog/watermarkpodautoscaler/blob/feat/wpa-cli/docs/kubectl-plugin.md 

I have also create the `krew` plugin configuratio, and updated the releasing tooling to also publish the plugin

**Review by CharlyF — APPROVED** [review]
great!

**Pull request merged**